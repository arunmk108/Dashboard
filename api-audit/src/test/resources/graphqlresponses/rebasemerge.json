{
    "data": {
        "repository": {
            "pullRequests": {
                "totalCount": 1,
                "pageInfo": {
                    "endCursor": "Y3Vyc29yOnYyOpK5MjAxNy0xMi0xMlQxMzoxNToyNS0wNTowMM4ADTnx",
                    "hasNextPage": false
                },
                "edges": [
                    {
                        "cursor": "Y3Vyc29yOnYyOpK5MjAxNy0xMi0xMlQxMzoxNToyNS0wNTowMM4ADTnx",
                        "node": {
                            "number": 1,
                            "mergeable": "UNKNOWN",
                            "state": "MERGED",
                            "createdAt": "2017-12-12T18:15:16Z",
                            "updatedAt": "2017-12-12T18:15:25Z",
                            "mergedAt": "2017-12-12T18:15:25Z",
                            "baseRef": {
                                "target": {
                                    "oid": "abbe21353249014eeda459130be9bfb5a66994b5"
                                }
                            },
                            "baseRefName": "master",
                            "title": "pr from fork",
                            "mergeCommit": {
                                "oid": "716f74018129a5a3b2578b773f27bef0892c962a"
                            },
                            "headRepository": {
                                "name": "RebaseMerge",
                                "url": "https://mygithub.com/ghtwo/RebaseMerge"
                            },
                            "headRef": {
                                "target": {
                                    "oid": "7c70c98b6cd93dcab57700a45302c1f7048c637c"
                                }
                            },
                            "headRefName": "master",
                            "author": {
                                "login": "ghtwo",
                                "resourcePath": "/ghtwo"
                            },
                            "commits": {
                                "totalCount": 2,
                                "nodes": [
                                    {
                                        "commit": {
                                            "oid": "cc4f007049ad2110e57b0a74b37c051d907040e6",
                                            "parents": {
                                                "nodes": [
                                                    {
                                                        "oid": "baeee8676685fcf6b979b88a26f4f2c8c980c4ae"
                                                    }
                                                ]
                                            },
                                            "committedDate": "2017-12-12T18:14:48Z",
                                            "message": "Third fork commit",
                                            "status": null,
                                            "author": {
                                                "name": "Author, Two",
                                                "date": "2017-12-12T13:14:48-05:00",
                                                "user": {
                                                    "login": "ghtwo"
                                                }
                                            }
                                        }
                                    },
                                    {
                                        "commit": {
                                            "oid": "7c70c98b6cd93dcab57700a45302c1f7048c637c",
                                            "parents": {
                                                "nodes": [
                                                    {
                                                        "oid": "cc4f007049ad2110e57b0a74b37c051d907040e6"
                                                    }
                                                ]
                                            },
                                            "committedDate": "2017-12-12T18:15:01Z",
                                            "message": "Fourth fork commit",
                                            "status": null,
                                            "author": {
                                                "name": "Author, Two",
                                                "date": "2017-12-12T13:15:01-05:00",
                                                "user": {
                                                    "login": "ghtwo"
                                                }
                                            }
                                        }
                                    }
                                ]
                            },
                            "comments": {
                                "totalCount": 0,
                                "nodes": []
                            },
                            "reviews": {
                                "totalCount": 0,
                                "nodes": []
                            }
                        }
                    }
                ]
            },
            "ref": {
                "target": {
                    "history": {
                        "pageInfo": {
                            "endCursor": "abbe21353249014eeda459130be9bfb5a66994b5 5",
                            "hasNextPage": false
                        },
                        "edges": [
                            {
                                "cursor": "abbe21353249014eeda459130be9bfb5a66994b5 0",
                                "node": {
                                    "oid": "abbe21353249014eeda459130be9bfb5a66994b5",
                                    "parents": {
                                        "nodes": [
                                            {
                                                "oid": "00d68be2db32072582f61300b60ae616b2b44b96"
                                            }
                                        ]
                                    },
                                    "message": "Sixth direct commit",
                                    "committer": {
                                        "user": null,
                                        "name": "GitHub Enterprise",
                                        "date": "2017-12-12T13:16:00-05:00"
                                    },
                                    "author": {
                                        "name": "Author, Two",
                                        "user": {
                                            "login": "ghtwo"
                                        },
                                        "email": "Author.Two@mycompany.com",
                                        "date": "2017-12-12T13:16:00-05:00"
                                    },
                                    "status": null
                                }
                            },
                            {
                                "cursor": "abbe21353249014eeda459130be9bfb5a66994b5 1",
                                "node": {
                                    "oid": "00d68be2db32072582f61300b60ae616b2b44b96",
                                    "parents": {
                                        "nodes": [
                                            {
                                                "oid": "716f74018129a5a3b2578b773f27bef0892c962a"
                                            }
                                        ]
                                    },
                                    "message": "Fifth direct commit",
                                    "committer": {
                                        "user": null,
                                        "name": "GitHub Enterprise",
                                        "date": "2017-12-12T13:15:47-05:00"
                                    },
                                    "author": {
                                        "name": "Author, Two",
                                        "user": {
                                            "login": "ghtwo"
                                        },
                                        "email": "Author.Two@mycompany.com",
                                        "date": "2017-12-12T13:15:47-05:00"
                                    },
                                    "status": null
                                }
                            },
                            {
                                "cursor": "abbe21353249014eeda459130be9bfb5a66994b5 2",
                                "node": {
                                    "oid": "716f74018129a5a3b2578b773f27bef0892c962a",
                                    "parents": {
                                        "nodes": [
                                            {
                                                "oid": "ff57d77108231a59bff431ec51a13369df4b314d"
                                            }
                                        ]
                                    },
                                    "message": "Fourth fork commit",
                                    "committer": {
                                        "user": {
                                            "login": "ghtwo"
                                        },
                                        "name": "Author, Two",
                                        "date": "2017-12-12T13:15:24-05:00"
                                    },
                                    "author": {
                                        "name": "Author, Two",
                                        "user": {
                                            "login": "ghtwo"
                                        },
                                        "email": "Author.Two@mycompany.com",
                                        "date": "2017-12-12T13:15:01-05:00"
                                    },
                                    "status": null
                                }
                            },
                            {
                                "cursor": "abbe21353249014eeda459130be9bfb5a66994b5 3",
                                "node": {
                                    "oid": "ff57d77108231a59bff431ec51a13369df4b314d",
                                    "parents": {
                                        "nodes": [
                                            {
                                                "oid": "baeee8676685fcf6b979b88a26f4f2c8c980c4ae"
                                            }
                                        ]
                                    },
                                    "message": "Third fork commit",
                                    "committer": {
                                        "user": {
                                            "login": "ghtwo"
                                        },
                                        "name": "Author, Two",
                                        "date": "2017-12-12T13:15:24-05:00"
                                    },
                                    "author": {
                                        "name": "Author, Two",
                                        "user": {
                                            "login": "ghtwo"
                                        },
                                        "email": "Author.Two@mycompany.com",
                                        "date": "2017-12-12T13:14:48-05:00"
                                    },
                                    "status": null
                                }
                            },
                            {
                                "cursor": "abbe21353249014eeda459130be9bfb5a66994b5 4",
                                "node": {
                                    "oid": "baeee8676685fcf6b979b88a26f4f2c8c980c4ae",
                                    "parents": {
                                        "nodes": [
                                            {
                                                "oid": "b205588963b31108414cfd925d2ebcbb0041ef1c"
                                            }
                                        ]
                                    },
                                    "message": "Second direct commit",
                                    "committer": {
                                        "user": null,
                                        "name": "GitHub Enterprise",
                                        "date": "2017-12-12T13:14:14-05:00"
                                    },
                                    "author": {
                                        "name": "Author, Two",
                                        "user": {
                                            "login": "ghtwo"
                                        },
                                        "email": "Author.Two@mycompany.com",
                                        "date": "2017-12-12T13:14:14-05:00"
                                    },
                                    "status": null
                                }
                            },
                            {
                                "cursor": "abbe21353249014eeda459130be9bfb5a66994b5 5",
                                "node": {
                                    "oid": "b205588963b31108414cfd925d2ebcbb0041ef1c",
                                    "parents": {
                                        "nodes": []
                                    },
                                    "message": "First direct commit",
                                    "committer": {
                                        "user": null,
                                        "name": "GitHub Enterprise",
                                        "date": "2017-12-12T13:13:58-05:00"
                                    },
                                    "author": {
                                        "name": "Author, Two",
                                        "user": {
                                            "login": "ghtwo"
                                        },
                                        "email": "Author.Two@mycompany.com",
                                        "date": "2017-12-12T13:13:58-05:00"
                                    },
                                    "status": null
                                }
                            }
                        ]
                    }
                }
            },
            "issues": {
                "totalCount": 0,
                "pageInfo": {
                    "endCursor": null,
                    "hasNextPage": false
                },
                "edges": []
            }
        }
    }
}