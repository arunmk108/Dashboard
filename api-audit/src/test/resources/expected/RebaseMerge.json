[
    {
        "auditStatuses": [
            "GIT_FORK_STRATEGY",
            "COMMITAUTHOR_EQ_MERGECOMMITER",
            "PULLREQ_NOT_PEER_REVIEWED"
        ],
        "lastUpdated": 1516207280689,
        "pullRequest": {
            "scmUrl": "https://mygithub.com/Devopscode/RebaseMerge",
            "scmBranch": "master",
            "scmRevisionNumber": "716f74018129a5a3b2578b773f27bef0892c962a",
            "scmCommitLog": "pr from fork",
            "scmCommitTimestamp": 1513102525000,
            "numberOfChanges": 0,
            "id": "5a5f77a16fa3ba5285b97546",
            "orgName": "Devopscode",
            "repoName": "RebaseMerge",
            "sourceRepo": "ghtwo/RebaseMerge",
            "sourceBranch": "master",
            "targetRepo": "Devopscode/RebaseMerge",
            "targetBranch": "master",
            "number": "1",
            "collectorItemId": "5a4e54d833ec296b81c39395",
            "updatedAt": 1513102525000,
            "createdAt": 1513102516000,
            "closedAt": 0,
            "state": "merged",
            "mergedAt": 1513102525000,
            "timestamp": 1513102516000,
            "resolutiontime": 0,
            "userId": "ghtwo",
            "comments": [],
            "reviews": [],
            "headSha": "7c70c98b6cd93dcab57700a45302c1f7048c637c",
            "baseSha": "abbe21353249014eeda459130be9bfb5a66994b5",
            "requestType": "pull",
            "commits": [
                {
                    "scmRevisionNumber": "cc4f007049ad2110e57b0a74b37c051d907040e6",
                    "scmCommitLog": "Third fork commit",
                    "scmAuthor": "Author, Two",
                    "scmAuthorLogin": "ghtwo",
                    "scmParentRevisionNumbers": [
                        "baeee8676685fcf6b979b88a26f4f2c8c980c4ae"
                    ],
                    "scmCommitTimestamp": 1513102488000,
                    "numberOfChanges": 0,
                    "timestamp": 0,
                    "firstEverCommit": false
                },
                {
                    "scmRevisionNumber": "7c70c98b6cd93dcab57700a45302c1f7048c637c",
                    "scmCommitLog": "Fourth fork commit",
                    "scmAuthor": "Author, Two",
                    "scmAuthorLogin": "ghtwo",
                    "scmParentRevisionNumbers": [
                        "cc4f007049ad2110e57b0a74b37c051d907040e6"
                    ],
                    "scmCommitTimestamp": 1513102501000,
                    "numberOfChanges": 0,
                    "timestamp": 0,
                    "firstEverCommit": false
                }
            ]
        },
        "commits": [
            {
                "scmRevisionNumber": "cc4f007049ad2110e57b0a74b37c051d907040e6",
                "scmCommitLog": "Third fork commit",
                "scmAuthor": "Author, Two",
                "scmAuthorLogin": "ghtwo",
                "scmParentRevisionNumbers": [
                    "baeee8676685fcf6b979b88a26f4f2c8c980c4ae"
                ],
                "scmCommitTimestamp": 1513102488000,
                "numberOfChanges": 0,
                "timestamp": 0,
                "firstEverCommit": false
            },
            {
                "scmRevisionNumber": "7c70c98b6cd93dcab57700a45302c1f7048c637c",
                "scmCommitLog": "Fourth fork commit",
                "scmAuthor": "Author, Two",
                "scmAuthorLogin": "ghtwo",
                "scmParentRevisionNumbers": [
                    "cc4f007049ad2110e57b0a74b37c051d907040e6"
                ],
                "scmCommitTimestamp": 1513102501000,
                "numberOfChanges": 0,
                "timestamp": 0,
                "firstEverCommit": false
            }
        ],
        "scmUrl": "https://mygithub.com/Devopscode/RebaseMerge",
        "scmBranch": "master"
    },
    {
        "auditStatuses": [
            "SCM_AUTHOR_LOGIN_INVALID",
            "DIRECT_COMMITS_TO_BASE_FIRST_COMMIT",
            "DIRECT_COMMITS_TO_BASE"
        ],
        "lastUpdated": 1516207280689,
        "commits": [
            {
                "scmUrl": "https://mygithub.com/Devopscode/RebaseMerge",
                "scmBranch": "master",
                "scmRevisionNumber": "abbe21353249014eeda459130be9bfb5a66994b5",
                "scmCommitLog": "Sixth direct commit",
                "scmAuthor": "Author, Two",
                "scmAuthorLogin": "ghtwo",
                "scmParentRevisionNumbers": [
                    "00d68be2db32072582f61300b60ae616b2b44b96"
                ],
                "scmCommitTimestamp": 1513102560000,
                "numberOfChanges": 1,
                "type": "New",
                "id": "5a5f77a16fa3ba5285b97540",
                "collectorItemId": "5a4e54d833ec296b81c39395",
                "timestamp": 1516205985496,
                "firstEverCommit": false
            },
            {
                "scmUrl": "https://mygithub.com/Devopscode/RebaseMerge",
                "scmBranch": "master",
                "scmRevisionNumber": "00d68be2db32072582f61300b60ae616b2b44b96",
                "scmCommitLog": "Fifth direct commit",
                "scmAuthor": "Author, Two",
                "scmAuthorLogin": "ghtwo",
                "scmParentRevisionNumbers": [
                    "716f74018129a5a3b2578b773f27bef0892c962a"
                ],
                "scmCommitTimestamp": 1513102547000,
                "numberOfChanges": 1,
                "type": "New",
                "id": "5a5f77a16fa3ba5285b97541",
                "collectorItemId": "5a4e54d833ec296b81c39395",
                "timestamp": 1516205985496,
                "firstEverCommit": false
            },
            {
                "scmUrl": "https://mygithub.com/Devopscode/RebaseMerge",
                "scmBranch": "master",
                "scmRevisionNumber": "baeee8676685fcf6b979b88a26f4f2c8c980c4ae",
                "scmCommitLog": "Second direct commit",
                "scmAuthor": "Author, Two",
                "scmAuthorLogin": "ghtwo",
                "scmParentRevisionNumbers": [
                    "b205588963b31108414cfd925d2ebcbb0041ef1c"
                ],
                "scmCommitTimestamp": 1513102454000,
                "numberOfChanges": 1,
                "type": "New",
                "id": "5a5f77a16fa3ba5285b97544",
                "collectorItemId": "5a4e54d833ec296b81c39395",
                "timestamp": 1516205985496,
                "firstEverCommit": false
            },
            {
                "scmUrl": "https://mygithub.com/Devopscode/RebaseMerge",
                "scmBranch": "master",
                "scmRevisionNumber": "b205588963b31108414cfd925d2ebcbb0041ef1c",
                "scmCommitLog": "First direct commit",
                "scmAuthor": "Author, Two",
                "scmAuthorLogin": "ghtwo",
                "scmParentRevisionNumbers": [],
                "scmCommitTimestamp": 1513102438000,
                "numberOfChanges": 1,
                "type": "New",
                "id": "5a5f77a16fa3ba5285b97545",
                "collectorItemId": "5a4e54d833ec296b81c39395",
                "timestamp": 1516205985496,
                "firstEverCommit": true
            }
        ],
        "scmUrl": "https://mygithub.com/Devopscode/RebaseMerge",
        "scmBranch": "master"
    }
]